erpnext/erpnext_integrations/exotel_integration.py

```diff
@@ -1,25 +1,30 @@
 import frappe
 import requests
+from frappe import _
 
 # api/method/erpnext.erpnext_integrations.exotel_integration.handle_incoming_call
 # api/method/erpnext.erpnext_integrations.exotel_integration.handle_end_call
 # api/method/erpnext.erpnext_integrations.exotel_integration.handle_missed_call
 
 @frappe.whitelist(allow_guest=True)
 def handle_incoming_call(**kwargs):
-	exotel_settings = get_exotel_settings()
-	if not exotel_settings.enabled: return
-
-	call_payload = kwargs
-	status = call_payload.get('Status')
-	if status == 'free':
-		return
-
-	call_log = get_call_log(call_payload)
-	if not call_log:
-		create_call_log(call_payload)
-	else:
-		update_call_log(call_payload, call_log=call_log)
+	try:
+		exotel_settings = get_exotel_settings()
+		if not exotel_settings.enabled: return
+
+		call_payload = kwargs
+		status = call_payload.get('Status')
+		if status == 'free':
+			return
+
+		call_log = get_call_log(call_payload)
+		if not call_log:
+			create_call_log(call_payload)
+		else:
+			update_call_log(call_payload, call_log=call_log)
+	except Exception as e:
+		frappe.log_error(title=_('Error in Exotel incoming call'))
```

Comment:
First rollback, then log_error and commit

Reply:
@nabinhait Why do we require to rollback?
There's only one method which is trying to create a log... if it fails then or something fails before that, there will be nothing to rollback...

Reply:
Suppose, an error raised in [on_update](https://github.com/frappe/erpnext/blob/develop/erpnext/communication/doctype/call_log/call_log.py#L22) while creating call log. The call log has already been inserted in the database at that time. If we don't rollback, it will get committed via except block.
It is always a good practice, to do rollback in `except` if we are committing at the end of `try` block.